main/mariadb/APKBUILD

```diff
@@ -90,7 +90,7 @@ build() {
 		-DINSTALL_SUPPORTFILESDIR=share/$pkgname \
 		-DINSTALL_MYSQLSHAREDIR=share/$pkgname \
 		-DINSTALL_DOCDIR=share/doc/$pkgname \
-		-DTMPDIR=/var/tmp \
+		-DTMPDIR=/var/tmp/$pkgname \
```

Comment:
Why is `/var/tmp` wrong?

Reply:
I experienced an issue on a clean baremetal install with only mariadb, where mariadb (innodb) would not start. It turned out to be because caused by the mariadb user dit not have permissions to create its temp files.

I thought that I would be best to fix it by giving mariadb its own subdirectory.

Reply:
You must have had wrong permissions of `/var/tmp`. It is supposed to be writable by everybody. Must be a bug somewhere else.

Reply:
No, `/var/tmp` should not be writable for everybody, this applies only to `/tmp`! `/var/tmp/$pkgname` is correct.

Reply:
This is weird, alpine-baselayout really creates `/var/tmp` with 1777, but when I look at various Alpine installations I have, everywhere is `0755` (and I did not modify it). o.O

Reply:
I searched web and checked out few other distros, it seems that `/var/tmp` should be really world writable. But all of the Alpine systems I have, including these installed using Alpine scripts, have `/var/tmp` 0755. That's also what I remember.

I thought that the common practice for packaging is to create a subdirectory named after the package in `/var/tmp`, owned by the user the installed program run as and typically with permissions 0700. Maybe I learned this from Gentoo, don't remember.

Anyway, I'm convinced that this practice is better than letting programs write directly to `/var/tmp`. It's more secure, prevents conflicts between programs that may create the same named files/directories and avoids mess in `/var/tmp`.

Reply:
> No, /var/tmp is not and should not be writable for everybody

Do you have any documentation or anything to back up that claim? Are you sure that you are not confusing it with `/var/lib`, which should be writable by root only?

[We set it](https://git.alpinelinux.org/cgit/aports/tree/main/alpine-baselayout/APKBUILD#n110) to `1777`, similar to other distros.

According [FHS 3.0](http://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s15.html) the purpose is to keep temporary files similiar to `/tmp`, but with the difference that it persist across reboots.

Reply:
I already explained it in my [last comment](https://github.com/alpinelinux/aports/pull/4630#discussion_r213964170), you probably wrote your reply before I posted it.

Reply:
> This is weird, alpine-baselayout really creates /var/tmp with 1777, but when I look at various Alpine installations I have, everywhere is 0755 (and I did not modify it). o.O

This is why I say there is a bug somwhere else.

Reply:
> you probably wrote your reply before I posted it.

I did. sorry.

let us follow up this on https://bugs.alpinelinux.org/issues/9364